lib/tree.rb

```diff
@@ -1,64 +1,176 @@
 class TreeNode
   attr_reader :key, :value
   attr_accessor :left, :right
-
-   def initialize(key, val)
+  
+  def initialize(key, val)
     @key = key
     @value = val
     @left = nil
     @right = nil
-   end
+  end
 end
 
 class Tree
   attr_reader :root
   def initialize
     @root = nil
   end
-
-  # Time Complexity: 
-  # Space Complexity: 
+  
+  # Time Complexity: O(n) if the tree is unbalanced and resembles a linked list
+  # Space Complexity: O(1)
   def add(key, value)
```

Comment:
Since you're doing things recursively you have space complexity based on the system stack.  For a balanced tree the space complexity would be O(log n) and for an unbalanced tree O(n)